carbonmark-api/src/utils/helpers/fetchMarketplaceListings.ts

```diff
@@ -71,52 +73,74 @@ export const fetchMarketplaceListings = async ({
       : null,
   }));
 
-  const getListingsWithProfiles = Promise.all(
-    formattedListings.map(async (listing) => {
-      const seller = await fastify.firebase
-        .firestore()
-        .collection("users")
-        .doc(listing.seller.id.toUpperCase())
-        .get();
-      return {
-        ...listing,
-        seller: {
-          ...listing.seller,
-          ...seller.data(),
-        },
-      };
-    })
+  const userIds = new Set<string>();
+  formattedListings.forEach((listing) =>
+    userIds.add(listing.seller.id.toLowerCase())
   );
-  const getActivitiesWithProfiles: Promise<ActivityWithUserHandles[]> =
-    Promise.all(
-      formattedActivities.map(async (act) => {
-        const activityWithHandles: ActivityWithUserHandles = { ...act };
-        const seller = await fastify.firebase
+  formattedActivities.forEach((activity) => {
+    userIds.add(activity.seller.id.toLowerCase());
+    if (activity.buyer) {
+      userIds.add(activity.buyer.id.toLowerCase());
+    }
+  });
+
+  const usersById = new Map<string, DocumentData | undefined>();
+
+  if (userIds.size !== 0) {
+    const chunkSize = 30;
+    const chunks : string[][] = [];
+    const ids = Array.from(userIds);
+
+    for (let i = 0; i < ids.length; i += chunkSize) {
+      chunks.push(ids.slice(i, i + chunkSize));
+    }
```

Comment:
We could make use of lodash's `chunk` function here: https://lodash.com/docs/4.17.15#chunk